src/openforms/prefill/api/serializers.py

```diff
@@ -17,6 +24,46 @@ class PrefillPluginSerializer(PluginBaseSerializer):
     )
 
 
+PrefillPluginPredicate = Callable[[BasePlugin], bool]
+
+
+class PrefillPluginPredicateSerializer(serializers.Serializer[PrefillPluginPredicate]):
```

Comment:
I probably didn't make myself clear enough, so to make it explicit and unambiguous:

1. Drop the `match` / `case`, there is literally only a single filter field to consider
2. implement the filtering exactly like the `ValidatorsListView` and `ValidatorsFilterSerializer` do
3. Update `ValidatorsFilterSerializer` to apply the camelCase -> snake_case conversion like this serializer here does. 

There's only a single filter parameter to consider and the only job of the serializer in a query/filter parameter context is input validation (and camel -> underscore conversion, but for that I need to check how/if we handled that in Open Zaak. Possibly we can extend the DRF parser to apply this conversion). The pattern matching is an over-engineered solution that only handles a single filter paramter, especially considering we'd rather not have suwinet be a prefill plugin in the first place so if we achieve that, this entire filtering feature can be deleted again.

I have a couple more thoughts I want to discuss in the office tomorrow because I'm failing to follow the reasoning behind decisions and I don't want to assume anything or drag this out in a public repo/PR.